File: user/src/main/java/com/vasanth/user/service/UserService.java
package com.vasanth.user.service;

import com.vasanth.user.model.User;
import com.vasanth.user.shared.Authentication;
import com.vasanth.user.shared.Credentials;

public interface UserService {
    public void saveUser(User user);
    public Authentication verify(Credentials credentials);
}

File: frontend/src/app/service/user.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Credentials} from '../Credentials';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http: HttpClient) { }
  public verify(credential: Credentials): Observable<any>{
    return this.http.post('http://localhost:8083/user/verify', credential );
  }
}
